api/src/utils/task-queue.ts

```diff
@@ -0,0 +1,64 @@
+import fastq, { asyncWorker } from 'fastq';
+import { getLogger } from './logger';
+
+const defaultLog = getLogger('TaskQueue');
+
+export type QueueResult<T, J> =
+  | {
+      task: T;
+      value: J;
+      error?: never;
+    }
+  | {
+      task: T;
+      value?: never;
+      error: Error;
+    };
+
+/**
+ * Process a list of tasks by queueing them and resolving them in parallel concurrently.
+ *
+ * Conceptually equivalent to:
+ * @example
+ *  const taskArrays = [[promiseA, promiseB], [promiseC, promiseD]];
+ *  for (const tasks of taskArrays) {
+ *    const results = Promise.all(task)
+ *  }
+ *
+ * @template T - The type of the tasks to process via `asyncWorker`
+ * @template J - The type of the resolved value from `asyncWorker`
+ * @param {T[]} tasks - The tasks to process
+ * @param {asyncWorker<unknown, T, J>} asyncWorker - The worker function that processes each task
+ * @param {number} concurrently - The number of tasks to process concurrently
+ * @returns {Promise<QueueResult<T,J>[]>}
+ */
+export const taskQueue = async <T, J>(
```

Comment:
nit: Might be nice to change `T` and `J` to something more descriptive.  Like `TaskType` and `TaskResponseType` or something.

Reply:
Ok I updated them to TaskType + WorkerResultType